src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -994,10 +996,14 @@ public long GetLastMappedEtagFor(string collection)
             }
         }
 
+        private NativeMemory.ThreadStats _indexingThreadStats; 
```

Comment:
What is this for?

Reply:
To hold a reference to the indexing stats for the indexing thread.
We can call `ReduceMemoryUsage` from outside of the indexing thread, but we need to get the stats of the index.